### PR TITLE
Update examples to use California housing dataset

### LINKED ISSUE #490: Replace Boston dataset in examples
The Boston dataset which we use in some examples has an ethical problem and should be replaced. Read more here: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html#sklearn.datasets.load_boston

Impacted examples:
 - `cfproto_housing.ipynb`
 - `ale_regression_boston.ipynb`

The above link suggests some similar housing-related alternatives.

### PR DESCRIPTION
**Pull request by jklaise**
Fixes #490.

Todo:
- [x] `cfproto_housing.ipynb`
- [x] `ale_regression_boston.ipynb`  -> `ale_regression_california.ipynb`
- [x] Update links

### CODE REVIEW

**Comment by jklaise** [comment]
Note that due to an unfortunate bug in `jupyter` regarding dollar sign escaping there is some slight weirdness with extra lines in the paragraph describing the effect of median income - due to `jupyter` not being able to render more than one dollar sign in the same paragraph: https://github.com/jupyterlab/jupyterlab/issues/8645

File: doc/source/examples/ale_regression_california.ipynb [hunk 1]
```diff
@@ -0,0 +1,875 @@
+{
```
Comment (RobertSamoilescu) [inline doc/source/examples/ale_regression_california.ipynb:1]: <p>I will probably mention explicitly that we are talking about average prediction of the data points which have their <code>MedInc</code> value within that bin.</p><p> </p><p>interpeted -&gt; interpreted</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (RobertSamoilescu) [inline doc/source/examples/ale_regression_california.ipynb:1]: <p>In the last sentence 'a smaller median income in the are bring the predicted house value down with respect to the average prediction.' shouldn't it be ' ... average feature effect'?</p><p> </p><p>in the are bring -&gt; in the area brings</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (RobertSamoilescu) [inline doc/source/examples/ale_regression_california.ipynb:1]: <p>I would say that the deciles don't show where there isn't any data, but is more like an indication of lower density. Take for example the plot of <code>MedInc</code>. In the last decile there exist data points, it's just that there is lower density of points.  For the example below, the statement might actually be true. But the absence of points I think is illustrated by the absence of markers on the lineplot, and not by the deciles.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (RobertSamoilescu) [inline doc/source/examples/ale_regression_california.ipynb:1]: <p>I don't think you need the assumption on the conditional feature distribution for the expression to be true.</p><p> </p><p>learnt -&gt; learned ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Comment (RobertSamoilescu) [inline doc/source/examples/ale_regression_california.ipynb:1]: <p>There are some empty axes. Not a big deal, but would be great to remove them.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

File: doc/source/examples/cfproto_housing.ipynb [hunk 2]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```
Comment (RobertSamoilescu) [inline doc/source/examples/cfproto_housing.ipynb:4]: <p>pX_testtype -&gt; prototype?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi/pull/838/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

File: doc/source/examples/ale_regression_california.ipynb (hunk 1 above)
Comment (jklaise) [reply doc/source/examples/ale_regression_california.ipynb:1]: <p>Removed the assumption as I think you're right, in any case the sentence didn't clarify much.</p>

Comment (jklaise) [reply doc/source/examples/ale_regression_california.ipynb:1]: <p>Yup, forgot to update.</p>

Comment (jklaise) [reply doc/source/examples/ale_regression_california.ipynb:1]: <p>Agreed, rephrased.</p>

Comment (jklaise) [reply doc/source/examples/ale_regression_california.ipynb:1]: <p>Added a Note box explaining this.</p>

**Review by RobertSamoilescu — APPROVED** [review]

**Pull request merged**